flock_agent/daemon/api_client.py

```diff
@@ -129,25 +129,27 @@ def _make_request(self, path, method, auth, data=None):
         if res.status_code == 401:
             raise PermissionDenied()
 
-        if res.status_code != 200:
-            raise BadStatusCode(res)
+        if res.status_code == 200 or res.status_code == 400:
+            if res.content:
```

Comment:
If there's no content but the status_code is 200 it looks like this will return None, which might be unexpected by things calling it. But that's an existing maybe-bug.

Reply:
Hmm yeah. The problem is, `GET /ping` is supposed to respond with status code 200, and right now responds with no content. So if I make it check raise an `InvalidResponse` exception if there's no content, then the ping test fails. There are different ways of solving this, like the ping response can include content. But for now I'm ok with just merging this.